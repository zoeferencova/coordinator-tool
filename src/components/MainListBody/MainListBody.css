
.table {
    width: 100%;
    display: table;
    margin: 0 auto;
}

.table div {
    padding: 8px 0;
    font-size: 14px;
}

.table-header {
    display: table-header-group;
}

.table-header div {
    background-color: lightgray;
    font-weight: bold;
}

.table-header-cell {
    display: table-cell;
    padding: 10px;
    text-align: center;
    border-bottom: 1px solid black;
}

.table-body {
    display: table-row-group;
}

.table-row {
    display: table-row;
}

.table-body-cell {
    display: table-cell;
}

.table-body-cell button {
    padding: 2px;
    background-color: transparent;
    border: none;
}

#status {
    width: 20px;
}

.hide-desktop {
    display: none !important;
}


@media only screen and (max-width: 700px) {
    .hide-mobile {
        display: none !important;
    }

    .hide-desktop {
        display: block !important;
    }
    
    .table-row div {
        font-size: 14px;
    }

    .proj-cell, .adv-cell {        
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
    }

    .proj-cell {
        max-width: 100px;
    }

    .adv-cell {
        max-width: 70px;
    }

    .table-row i {
        font-size: 16px;
        margin: 1px;
    }

    .status-icon i {
        font-size: 20px;
    }

    
}


