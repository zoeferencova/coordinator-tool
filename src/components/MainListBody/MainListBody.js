import React from 'react';
import './MainListBody.css'
import MainListItem from '../MainListItem/MainListItem';

const dummyData = [
    {
        id: 1,
        checked: true,
        status: 'none',
        project: 'Fake Project',
        advisor: 'Zoe Ferencova',
        pm: 1,
        date: new Date().toLocaleDateString(),
        notes: 'Lorem ipsum'
    },
    {
        id: 2,
        checked: false,
        status: 'none',
        project: 'Dummy Project',
        advisor: 'James Park',
        pm: 2,
        date: new Date().toLocaleDateString(),
        notes: 'Dolor sit amet'
    },
    {
        id: 3,
        checked: false,
        status: 'none',
        project: 'Super Fake Project',
        advisor: 'Robin Hurst',
        pm: 3,
        date: new Date().toLocaleDateString(),
        notes: 'Consectetur adipiscing'
    },
    {
        id: 3,
        checked: false,
        status: 'none',
        project: 'Fake Project',
        advisor: 'Robin Hurst',
        pm: 3,
        date: new Date().toLocaleDateString(),
        notes: 'Consectetur adipiscing'
    },
    {
        id: 4,
        checked: false,
        status: 'none',
        project: 'Super Fake Project',
        advisor: 'Sara Waldman',
        pm: 3,
        date: new Date().toLocaleDateString(),
        notes: 'Consectetur adipiscing'
    },
    {
        id: 5,
        checked: false,
        status: 'none',
        project: 'Not Real Project',
        advisor: 'Marnie Hurst',
        pm: 2,
        date: new Date().toLocaleDateString(),
        notes: 'Consectetur adipiscing'
    },
    {
        id: 6,
        checked: false,
        status: 'none',
        project: 'Another Fake Project',
        advisor: 'Daniela Narvaez',
        pm: 1,
        date: new Date().toLocaleDateString(),
        notes: 'Consectetur adipiscing'
    },
]

export default class MainListBody extends React.Component {   
    renderRows() {
        return dummyData.map(row => 
            <MainListItem
                key={row.id}
                checked={row.checked}
                status={row.status}
                project={row.project}
                advisor={row.advisor}
                pm={row.pm}
                date={row.date}
                notes={row.notes}
            />
        )
    }
    
    render() {
        return (
            <div className="table">
                <div className="table-header">
                    <div className="table-header-cell"><input type="checkbox"></input></div>
                    <div className="table-header-cell">Status</div>
                    <div className="table-header-cell">Project</div>
                    <div className="table-header-cell">Advisor</div>
                    <div className="table-header-cell">PM</div>
                    <div className="table-header-cell">Date</div>
                    <div className="table-header-cell">Notes</div>
                    <div className="table-header-cell">Actions</div>
                </div>
                {this.renderRows()}
            </div>
        ) 
    }
}