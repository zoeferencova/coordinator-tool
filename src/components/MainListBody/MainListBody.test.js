import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MainListBody from './MainListBody'

describe(`MainListBody component`, () => {
  it('renders a .MainListBody by default', () => {
    const wrapper = shallow(<MainListBody />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})