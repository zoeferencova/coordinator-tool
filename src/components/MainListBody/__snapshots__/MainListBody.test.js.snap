// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainListBody component renders a .MainListBody by default 1`] = `
<div
  className="table"
>
  <div
    className="tableHeader"
  >
    <div
      className="tableHeaderCell hideMobile check"
    >
      <input
        id="header-checkbox"
        onChange={[Function]}
        type="checkbox"
      />
    </div>
    <div
      className="tableHeaderCell project"
    >
      Project 
      <i
        className="fas fa-sort"
        onClick={[Function]}
      />
    </div>
    <div
      className="tableHeaderCell advisor"
    >
      Advisor 
      <i
        className="fas fa-sort"
        onClick={[Function]}
      />
    </div>
    <div
      className="tableHeaderCell hideMobile"
    >
      PM 
      <i
        className="fas fa-sort"
        onClick={[Function]}
      />
    </div>
    <div
      className="tableHeaderCell hideMobile"
    >
      Date 
      <i
        className="fas fa-sort"
        onClick={[Function]}
      />
    </div>
    <div
      className="tableHeaderCell hideMobile"
    >
      Notes
    </div>
    <div
      className="tableHeaderCell status"
    >
      Status 
      <i
        className="fas fa-sort"
        onClick={[Function]}
      />
    </div>
    <div
      className="tableHeaderCell actions hideMobile"
    >
      Actions
    </div>
    <div
      className="tableHeaderCell mobileActions hideDesktop"
    />
  </div>
</div>
`;
